2.3_control_flow.ipynb

```diff
@@ -157,29 +142,25 @@
     "  }\n",
     "}\n",
     "\n",
-    "// verify that the max of the three inputs is correct\n",
-    "class Max3Tester(c: Max3) extends PeekPokeTester(c) {\n",
-    "  poke(c.io.in1, 6)\n",
-    "  poke(c.io.in2, 4)  \n",
-    "  poke(c.io.in3, 2)  \n",
-    "  expect(c.io.out, 6)  // input 1 should be biggest\n",
-    "  poke(c.io.in2, 7)  \n",
-    "  expect(c.io.out, 7)  // now input 2 is\n",
-    "  poke(c.io.in3, 11)  \n",
-    "  expect(c.io.out, 11) // and now input 3\n",
-    "  poke(c.io.in3, 3)  \n",
-    "  expect(c.io.out, 7)  // show that decreasing an input works as well\n",
-    "  poke(c.io.in1, 9)\n",
-    "  poke(c.io.in2, 9)\n",
-    "  poke(c.io.in3, 6)\n",
-    "  expect(c.io.out, 9)  // still get max with tie\n",
-    "}\n",
-    "\n",
     "// Test Max3\n",
-    "val works = Driver(() => new Max3) {\n",
-    "  c => new Max3Tester(c)\n",
+    "test(new Max3) { c =>\n",
```

Comment:
Similarly, this could be a useful example to demonstrate writing a test helper

Reply:
It would be. Again, I don't have time right now and I am going to be doing a bootcamp in a week. If I have time I will try and make a helper, is there decent docs somewhere on how to go about that

Reply:
I tried this as an exercise and I had a lot of trouble getting it to work.
I tried the following in several different ways and no matter what I did I got errors on trying to create hardware outside of a module

```    def sortTestHelper( c: Sort4, values: Array[UInt], sortAscending: Boolean = true): Unit = {
      assert(values.length == 4)
      c.io.in0 := values(0)
      c.io.in1 := values(1)
      c.io.in2 := values(2)
      c.io.in3 := values(3)

      val expected = values.sortBy(n => n.litValue())
      c.io.out0.expect(expected(0))
      c.io.out1.expect(expected(1))
      c.io.out2.expect(expected(2))
      c.io.out3.expect(expected(3))
    }
    
  sortTestHelper(c, Array(3.U, 6.U, 9.U, 12.U))
```